pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -0,0 +1,143 @@
+		},
+	}
+
+	mutateFunc := func() error {
+
+		// labels for the pod, the deployment, and the deploymentSelector
+		deploymentLabels := map[string]string{
+			corev1.LabelHostname: nodeHostnameLabel,
+			k8sutil.AppAttr:      AppName,
+			NodeNameLabel:        node.GetName(),
+		}
+
+		nodeSelector := map[string]string{corev1.LabelHostname: nodeHostnameLabel}
+
+		// Deployment selector is immutable so we set this value only if
+		// a new object is going to be created
+		if deploy.ObjectMeta.CreationTimestamp.IsZero() {
+			deploy.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			}
+		}
+
+		deploy.ObjectMeta.Labels = deploymentLabels
+		deploy.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{Labels: deploymentLabels},
+			Spec: corev1.PodSpec{
+				NodeSelector: nodeSelector,
+				InitContainers: []corev1.Container{
+					getCrashDirInitContainer(cephCluster),
+					getCrashChownInitContainer(cephCluster),
+				},
+				Containers: []corev1.Container{
+					getCrashDaemonContainer(cephCluster),
+				},
+				// Tolerations:   tolerations,
+				RestartPolicy: corev1.RestartPolicyAlways,
+				HostNetwork:   cephCluster.Spec.Network.IsHost(),
+				Volumes:       opspec.DaemonVolumesBase(config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath), mon.KeyringStoreName),
+			},
+		}
+
+		return nil
+	}
+
+	return controllerutil.CreateOrUpdate(context.TODO(), r.client, deploy, mutateFunc)
+}
+
+func getCrashDirInitContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	crashPostedDir := path.Join(dataPathMap.ContainerCrashDir, "posted")
+
+	container := corev1.Container{
+		Name: "make-container-crash-dir",
+		Command: []string{
+			"mkdir",
+			"-p",
+		},
+		Args: []string{
+			crashPostedDir,
+		},
+		Image:           cephCluster.Spec.CephVersion.Image,
+		SecurityContext: mon.PodSecurityContext(),
+	}
+	return container
+}
+
+func getCrashChownInitContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+
+	return opspec.ChownCephDataDirsInitContainer(
+		*dataPathMap,
+		cephCluster.Spec.CephVersion.Image,
+		opspec.DaemonVolumeMounts(dataPathMap, mon.KeyringStoreName),
+		corev1.ResourceRequirements{},
+		&corev1.SecurityContext{},
+	)
+}
+
+func getCrashDaemonContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	cephImage := cephCluster.Spec.CephVersion.Image
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	crashEnvVar := corev1.EnvVar{Name: "CEPH_ARGS", Value: "-m $(ROOK_CEPH_MON_HOST) -k /etc/ceph/keyring-store/keyring -n client.admin"}
+	envVars := append(opspec.DaemonEnvVars(cephImage), crashEnvVar)
+
+	container := corev1.Container{
```

Comment:
Should we set some resource limits? I thought @travisn mentioned that some environments require resource limits to be specified.